src/dbnode/storage/index/convert/convert_test.go

```diff
@@ -87,18 +122,52 @@ func TestFromSeriesIDAndTagIterValid(t *testing.T) {
 	)
 	d, err := convert.FromSeriesIDAndTagIter(id, ident.NewTagsIterator(tags))
 	assert.NoError(t, err)
-	assert.Equal(t, "foo", string(d.ID))
-	assert.Len(t, d.Fields, 1)
-	assert.Equal(t, "bar", string(d.Fields[0].Name))
-	assert.Equal(t, "baz", string(d.Fields[0].Value))
+	assertContentsMatch(t, id, tags.Values(), d)
+}
+
+func TestFromSeriesIDAndTagIterReuseBytesFromSeriesId(t *testing.T) {
+	tests := []struct {
+		name string
+		id   string
+	}{
+		{
+			name: "tags in ID",
+			id:   "bar=baz,quip=quix",
+		},
+		{
+			name: "tags in ID with specific format",
+			id:   `{bar="baz",quip="quix"}`,
```

Comment:
Nit: add one more case with tags in reverse order.

Reply:
https://github.com/m3db/m3/pull/3075/commits/6bb6d9b40510ae5404955846d12c5319a9dc0a9e#diff-4b7b04cd8028a198a21da071cd0a8bbcc9d2f7fd0f6f4bd896a73019eccdfdfeR95-R98